File: mobile/app/src/main/java/com/resuera/eventhive/features/events/EventsApi.kt
package com.resuera.eventhive.features.events

import okhttp3.MultipartBody
import okhttp3.RequestBody
import retrofit2.Call
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.Multipart
import retrofit2.http.PATCH
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Part
import retrofit2.http.Path
import retrofit2.http.Query

interface EventsApi {

    @GET("/api/events")
    fun getAllEvents(): Call<List<EventResponse>>

    @GET("/api/events/my-events")
    fun getMyEvents(): Call<List<EventResponse>>

    @GET("/api/events/registered")
    fun getRegisteredEvents(): Call<List<EventResponse>>

    @Multipart
    @POST("/api/events")
    fun createEvent(
        @Part("title") title: RequestBody,
        @Part("description") description: RequestBody,
        @Part("startDate") startDate: RequestBody,
        @Part("endDate") endDate: RequestBody,
        @Part("location") location: RequestBody,
        @Part("category") category: RequestBody,
        @Part("maxParticipants") maxParticipants: RequestBody,
        @Part image: MultipartBody.Part?
    ): Call<EventResponse>

    @Multipart
    @PUT("/api/events/{id}")
    fun updateEvent(
        @Path("id") id: Long,
        @Part("title") title: RequestBody,
        @Part("description") description: RequestBody,
        @Part("startDate") startDate: RequestBody,
        @Part("endDate") endDate: RequestBody,
        @Part("location") location: RequestBody,
        @Part("category") category: RequestBody,
        @Part("maxParticipants") maxParticipants: RequestBody,
        @Part image: MultipartBody.Part?
    ): Call<EventResponse>

    @POST("/api/events/{id}/register")
    fun registerForEvent(@Path("id") eventId: Long): Call<EventResponse>

    @PATCH("/api/events/{id}/status")
    fun updateEventStatus(@Path("id") id: Long, @Query("status") status: String): Call<EventResponse>

    @DELETE("/api/events/{id}")
    fun deleteEvent(@Path("id") id: Long): Call<Void>
}

File: web/eventhive/src/features/events/eventsApi.ts
import { apiFetch } from '../../shared/api/client'
import type { EventItem } from '../../shared/types'

export interface EventRequest {
  title: string
  description: string
  startDate: string
  endDate: string
  location: string
  category: string
  maxParticipants: number
}

export const eventsApi = {
  getAll: () =>
    apiFetch<EventItem[]>('/api/events'),

  getMyEvents: () =>
    apiFetch<EventItem[]>('/api/events/my-events'),

  getRegistered: () =>
    apiFetch<EventItem[]>('/api/events/registered'),

  getByCategory: (category: string) =>
    apiFetch<EventItem[]>(`/api/events/category?category=${encodeURIComponent(category)}`),

  // JSON create (without image)
  create: (data: EventRequest) =>
    apiFetch<EventItem>('/api/events', {
      method: 'POST',
      body: data,
    }),

  // Multipart create (with optional image)
  createWithImage: (data: EventRequest, image?: File) => {
    const formData = new FormData()
    formData.append('title', data.title)
    formData.append('description', data.description)
    formData.append('startDate', data.startDate)
    formData.append('endDate', data.endDate)
    formData.append('location', data.location)
    formData.append('category', data.category)
    formData.append('maxParticipants', String(data.maxParticipants))
    if (image) formData.append('image', image)

    return apiFetch<EventItem>('/api/events', {
      method: 'POST',
      body: formData,
    })
  },

  update: (id: number, data: EventRequest, image?: File) => {
    const formData = new FormData()
    formData.append('title', data.title)
    formData.append('description', data.description)
    formData.append('startDate', data.startDate)
    formData.append('endDate', data.endDate)
    formData.append('location', data.location)
    formData.append('category', data.category)
    formData.append('maxParticipants', String(data.maxParticipants))
    if (image) formData.append('image', image)

    return apiFetch<EventItem>(`/api/events/${id}`, {
      method: 'PUT',
      body: formData,
    })
  },

  updateStatus: (id: number, status: string) =>
    apiFetch<EventItem>(`/api/events/${id}/status?status=${status}`, {
      method: 'PATCH',
    }),

  delete: (id: number) =>
    apiFetch<void>(`/api/events/${id}`, {
      method: 'DELETE',
    }),

  register: (id: number) =>
    apiFetch<EventItem>(`/api/events/${id}/register`, {
      method: 'POST',
    }),
}